src/storage/src/client.rs

```diff
@@ -313,6 +277,200 @@ pub(crate) mod info {
     }
 }
 
+pub struct ReadObject<'a> {
+    client: &'a Storage,
+    request: control::model::ReadObjectRequest,
+}
+
+impl<'a> ReadObject<'a> {
+    pub(crate) fn new(client: &'a Storage) -> Self {
+        ReadObject {
+            client,
+            request: control::model::ReadObjectRequest::new(),
+        }
+    }
+
+    /// Sets the value of [bucket][control::model::ReadObjectRequest::bucket].
+    pub fn set_bucket<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
+        self.request.bucket = v.into();
+        self
+    }
+
+    /// Sets the value of [object][control::model::ReadObjectRequest::object].
+    pub fn set_object<T: std::convert::Into<std::string::String>>(mut self, v: T) -> Self {
+        self.request.object = v.into();
+        self
+    }
+
+    /// Sets the value of [generation][control::model::ReadObjectRequest::generation].
+    pub fn set_generation<T: std::convert::Into<i64>>(mut self, v: T) -> Self {
+        self.request.generation = v.into();
+        self
+    }
```

Comment:
We need to start fleshing out the tests.  In generated code we do not test these functions. If am unsure because these functions are trivial (which recommends "no tests") and at the same time they are hand-crafted (which recommends "write a unit test").

Thoughts?

Reply:
Wrote unit tests in #2279 and after writing them, they feel mostly like clutter, especially since these are copy pasted over with 0 changes except in the doc comments.

More useful tests would make sure we set each of these and then see if they have the desired effect.

Reply:
> Wrote unit tests in #2279 and after writing them, they feel mostly like clutter, especially since these are copy pasted over with 0 changes except in the doc comments.

Feel free to close #2279 if that is the case.

> More useful tests would make sure we set each of these and then see if they have the desired effect.

By "desired effect" I think you mean "the right request message + query parameters + headers gets built / sent to GCS"? You could refactor the code to prepare the `reqwest::RequestBuilder` and then (I think?) you can examine the values.  That does seem more useful.